full/src/main/java/apoc/dv/JDBCResource.java

```diff
@@ -0,0 +1,94 @@
+package apoc.dv;
+
+import org.apache.commons.lang3.StringUtils;
+import org.neo4j.internal.helpers.collection.Pair;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.MatchResult;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+public class JDBCResource extends VirtualizedResource {
+
+    private final String queryParsed;
+
+    public JDBCResource(String name, Map<String, Object> config) {
+        super(Objects.requireNonNull(name, "Field `name` should be defined"),
+                (String) Objects.requireNonNull(config.get("url"), "Field `url` in config should be defined"),
+                (String) Objects.requireNonNull(config.get("desc"), "Field `desc` in config should be defined"),
+                (List<String>) Objects.requireNonNull(config.get("labels"), "Field `labels` in config should be defined"),
+                (String) Objects.requireNonNull(config.get("query"), "Field `query` in config should be defined"),
+                getParameters(config),
+                "JDBC");
+        this.queryParsed = parseQuery(config);
+    }
+
+    private static List<String> getParameters(Map<String, Object> config) {
+        final String query = (String) config.get("query");
+        final long questionMarks = countForQuestionMarks(query);
+        if (questionMarks > 0) {
+            return LongStream.range(0, questionMarks)
+                    .mapToObj(i -> "?")
+                    .collect(Collectors.toList());
+        }
+        return PLACEHOLDER_PATTERN.matcher(query).results()
+                .map(MatchResult::group)
+                .map(String::trim)
+                .filter(StringUtils::isNotBlank)
+                .collect(Collectors.toList());
+    }
+
+    private static String parseQuery(Map<String, Object> config) {
+        String query = (String) config.get("query");
+        return parseQuery(query);
+    }
+
+    private static String parseQuery(String query) {
+        final long questionMarks = countForQuestionMarks(query);
+        if (questionMarks > 0) {
+            return query;
+        }
+        return PLACEHOLDER_PATTERN.matcher(query).replaceAll("?");
```

Comment:
is it really needed? the query for a JDBC resource should follow the proper JDBC syntax for parameters, or?

Reply:
@jbarrasa asked for it